### ISSUE TITLE
I am confused

### ISSUE THREAD

**Issue by spotanjo3** [moonlightz/meo-epg#1]
To enable EPG to read this. Should I do this:

https://github.com/moonlightz/meo-epg/blob/master/meo.xml.gz

or

https://raw.githubusercontent.com/moonlightz/meo-epg/master/meo.xml.gz

**Comment by moonlightz** [comment]
Hi there. Since you started to write in English, I can do it as well... just for the sake for others to read this.

I am confused as well and I have no idea what are you trying to accomplish.
This was never meant to be a professional thing and it's provided as is. 
None of those links work obviously because when I refresh the xml files, I don't provide any compressed version.
I could provide .gz .tar.gz .zip versions along with the 2 big xml files. It's no use to me.
I could start to provide under request.

The objective of this project is to feed the tcl file (loaded with eggdrop bot). The xml files can be used for other things of course.

If you can, try to explain what are you trying to do.

**Comment by spotanjo3** [comment]
Hi. Thanks for replying. Sorry to confused you. i am telling you that I am using TiviMate to read the EPG for my MEO Portugal streaming. Dont know which link is right for EPG to read.

**Comment by moonlightz** [comment]
https://raw.githubusercontent.com/moonlightz/meo-epg/master/meo-modificado.xml should work. It's just a modified version of the bigger file but without radio stations.
https://raw.githubusercontent.com/moonlightz/meo-epg/master/meo.xml is the link for the bigger file.. just about 200 KB bigger.
If you need a .gz file, I can do that. Just need to add an extra instruction to the bat file for something that picks one of these xml and compresses it to .gz file.

**Comment by moonlightz** [comment]
Ok. Just been thinking about making .gz file for you and got astonished how much the gz file got. 20 megabyte file down to ~1.5 megabyte file.
https://raw.githubusercontent.com/moonlightz/meo-epg/master/meo.xml.gz should work.

**Comment by spotanjo3** [comment]
Thank you!

https://raw.githubusercontent.com/moonlightz/meo-epg/master/meo.xml.gz works. 

I noticed some Portuguese channels arent there. Some are missing.

**Comment by moonlightz** [comment]
That's correct. Some telecom operators like NOS, Vodafone, or NOWO have channels that MEO doesn't. MEO has some channels that the other three don't.
The channels you find inside of the xml are from the coax cable. Yet, there might be there some channels missing from the full set of channels listed on the STB. I don't have SPORT TV for example, nor ELEVEN SPORTS.
Perhaps there is a way to push the EPG from the MEO Go. When I use the mobile app or the Windows app, I can see the channels listed and there is some programme listings but no idea how to download that listing.
No idea that you can find a similar thing but for the NOS network. There was guy here on Github that provided a good small xml but he stopped providing updates for ages.
I believe it is impossible to download the EPG from the STB. The settings menu shows an ip and something named GPON but there's nothing to tell us how do we access that EPG from the remote server.

**Comment by moonlightz** [comment]
MEO messed up their EPG many weeks ago and I couldn't find a way to convert whatever they were using to a readable thing.
So I'm now providing a xml made directly from their server. 
The meo.xml now has 214 channels, including the sport channels. Give it a go if you want.

**Comment by KiSa04** [comment]
> So I'm now providing a xml made directly from their server.

So, you are now using their online guide API? (https://www.meo.pt/tv/canais-programacao/guia-tv)

**Comment by moonlightz** [comment]
No. But that site is probably related to the thing I'm using.
I do have something called MEO Go for Windows. 
I looked at the calls it makes and build a scripted tool that grabs all the channels it has available and the tool I created, generates a xml file from scratch and the xml file gets compressed.
The coax only provided a few more than 100 channels, now I can get 215 at this time of writing. On the other hand... it takes less than 2 minutes to get the whole EPG from coaxial, now it takes 10-13 minutes to gather all the 215 channels EPG starting yesterday until 6/7/8 days.
Essentially I looked what the Meo Go For Windows does and mimic'ed the behaviour. I could probably tweak something like having more programmes between fetchings.
I can probably look at that site you indicated and build something, if it provides more than 215 channels.

**Comment by KiSa04** [comment]
> grabs all the channels it has available and the tool I created, generates a xml file from scratch and the xml file gets compressed.
> The coax only provided a few more than 100 channels, now I can

well, using the site takes ~1h on my script, but i grab the program info as well (image and description of each program), which demands fetching each program (the channel EPG only returns the program names)

does MeoGO API return any program data apart from its name? i

**Comment by thomraider12** [comment]
> No. But that site is probably related to the thing I'm using. I do have something called MEO Go for Windows. I looked at the calls it makes and build a scripted tool that grabs all the channels it has available and the tool I created, generates a xml file from scratch and the xml file gets compressed. The coax only provided a few more than 100 channels, now I can get 215 at this time of writing. On the other hand... it takes less than 2 minutes to get the whole EPG from coaxial, now it takes 10-13 minutes to gather all the 215 channels EPG starting yesterday until 6/7/8 days. Essentially I looked what the Meo Go For Windows does and mimic'ed the behaviour. I could probably tweak something like having more programmes between fetchings. I can probably look at that site you indicated and build something, if it provides more than 215 channels.

Is that script in this repository?
(se quiser que fale português também posso)

**Comment by moonlightz** [comment]
Coax EPG returns channel names, title of the programme, start time, end time, description, and some other stuff like category. Not images. A common tv (smart or not) only will show the title name, some bar, description, category.

Meo Go for windows (you can download it using windows store) has a "Guia TV" (TV Guide), if you click on that, it will show a bunch of channels, the current programme, start time and end. If you click a channel, you can see a guide, you can see images. Meo Go also has scheduled programmes that you can rewatch, upto 7 days back, also with images. The logos, images, etc, are included with the "api". 

You can use nirshoft smartsniff to see the calls and you wll see pngs being downloaded separately.

It takes ~1 hour pulling all the stuff from the link above? WOW...

The EPG generated by me is just for some smallish purposes like some third-party bots or some tool. It already is over 20 megabyte file. With images tags and extras... it will explode over 30 MB or more. 
I can add the other tags like images, logos, etc to the xml file but... only if some people would need them there.

**Comment by moonlightz** [comment]
Eu até certo ponto, já tive o script em tcl neste repositório mas dias depois decidi retirar para impedir abusos. A altice/meo iria reparar em tanto flood e modificar o serviço para outra coisa. E também reparei que o url tem um id que é supostamente meu. Se fores bom a vasculhar o repositório git, podes reculperá-lo. Neste repositório, só estou a fornecer o xml compilado.

Se tcl não é a tua linguagem de programação preferida... é uma questão de instalares o Meo Go que ainda deve estar disponível na Windows Store ou Loja do Windows, ter o smartsniff e o driver que sniffa pacotes, abrir o Meo go, ir ao smartsniff e clicar no botão de play, clicar nos --- no meo go e clicar Guia Tv, o smartsniff deve mostrar muita linha. Pára o smartsniff e analisa o url enviado e os dados recebido, com a tua linguagem de programação preferida é uma questão de enviares esse url e analisar o que vem do servidor, o servidor manda tudo aos bocados. É a mesma treta com os canais.

Infelizmente não meti muitos comentários no script que fiz. Devo ter demorado 1 a 2 dias a ver como tudo funcionava e depois mais 1 a 3 dias para modificar o output que aparece na consola.

O output é este:

```
E:\MEOCAPTURA>tclsh -encoding utf-8 ott.tcl
A iniciar a recolha pelo servidor da meo...
215 canais ... Feito!
A adicionar os programas de televisão ...
 √ | X0001 | RTP1       | RTP1                      | 00:01.571 | 129 programas   - Feito!
 √ | X0002 | RTP2       | RTP2                      | 00:06.559 | 555 programas   - Feito!
 √ | X0003 | SIC        | SIC                       | 00:01.471 | 122 programas   - Feito!
(.......)
 √ | X0922 | BTVCAM3    | BTV CAM3                  | 00:00.738 | 49 programas    - Feito!
 √ | X0923 | BTVCAM4    | BTV CAM4                  | 00:00.941 | 49 programas    - Feito!
O tamanho do ficheiro meo.xml é 22571389 bytes.
A operação demorou 11:27.141 segundos!
```

**Comment by thomraider12** [comment]
Uau. Muito obrigado. Vou vasculhar e fazer em Python.
